pkg/service/manager.go

```diff
@@ -170,6 +175,21 @@ func (sm *Manager) Start() error {
 		cancel()
 	}()
 
+	// Before starting the leader Election enable any additional functionality
+	upnpEnabled, _ := strconv.ParseBool(os.Getenv("enableUPNP"))
+
+	if upnpEnabled {
+		sm.upnp = new(upnp.Upnp)
+		err := sm.upnp.ExternalIPAddr()
+		if err != nil {
+			log.Errorf("Error Enabling UPNP %s", err.Error())
+			// Set the struct to nil so nothing should use it in future
+			sm.upnp = nil
+		} else {
```

Comment:
> Errors will be hard to handle as it's not something that Kubernetes load-balancers are designed to handle, errors would exist the kube-vip pod in the namespace the load-balancer is created in. This is where an end-user would find that the exposed port hasn't made it to the router (such as having an already existing port-mapping created).

Why are we not returning the `err`?

Reply:
If the environment variable does't exist OR it isn't a string form of "yes/YES/true/TRUE" then the bool is set to `false` .. the error wouldn't be any use in this context.